ddtrace/tracer.py

```diff
@@ -96,6 +98,7 @@ def _default_span_processors_factory(
     appsec_enabled,  # type: bool
     compute_stats_enabled,  # type: bool
     agent_url,  # type: str
+    tracer,  # type: Tracer
```

Comment:
This method should be independent of a tracer instance. This smells like coupling between the tracer and processors due to `gateway`.

Is there a need for tracer to have a `gateway`? AFAICT the gateway could be global